### PR TITLE
Exclude ladder level on both student and teacher completeness check

### PR DESCRIPTION
**Pull request by AndrewJakubowicz**
Checked the following test cases manually and cases in the comments on Asana.

The teacher calculation remains unchanged and still shows 100%.
The student 18/19 correctly displays as 18/18 as ladder is not a required level.
This is correctly displayed on the /students page as well as the world map for the student.

### CODE REVIEW

**Comment by AndrewJakubowicz** [comment]
This fix initially rebroke the teacher. Incorrectly showing 18/19 on teacher when arena should be ignored.
This was fixed in the second commit by making ladder levels optional for the teacher calculation.

File: app/lib/coursesHelper.coffee [hunk 1]
```diff
@@ -178,7 +178,7 @@ module.exports =
             numStarted: 0
             # numCompleted: 0
           }
-          isOptional = level.get('practice') or level.get('assessment')
+          isOptional = level.get('practice') or level.get('assessment') or level.isLadder()
```
Comment (AndrewJakubowicz) [inline app/lib/coursesHelper.coffee]: This change filters out the `isLadder` levels on the teacher side which has parity with the student side as well as our decision that arenas are not required.

File: app/models/Classroom.coffee [hunk 2]
```diff
@@ -169,7 +171,7 @@ module.exports = class Classroom extends CocoModel
     nextLevel = courseLevels.models[nextIndex]
     nextLevel = arena if levelsLeft is 0
     nextLevel ?= _.find courseLevels.models, (level) -> not levelSessionMap[level.get('original')]?.get('state')?.complete
-    [userStarted, courseComplete, totalComplete] = coursesHelper.hasUserCompletedCourse(userLevels, levelsInCourse)
+    [_userStarted, courseComplete, _totalComplete] = coursesHelper.hasUserCompletedCourse(userLevels, levelsInCourse)
```
Comment (shubhi1092) [inline app/models/Classroom.coffee]: Why this change?

Comment (AndrewJakubowicz) [reply app/models/Classroom.coffee]: The underscores are to try and show that those variables are unused.

**Comment by AndrewJakubowicz** [comment]
There are some very light automated tests that test `courseHelper.hasUserCompletedCourse` but the rest is untested and appears very challenging to write tests for due to the immense size of the functions.
